include/argparser.h

```diff
@@ -0,0 +1,73 @@
+//
+// Created by Sebastian Schmitt on 19.03.2020.
+//
+
+#ifndef YAPGT_ARGPARSER_H
+#define YAPGT_ARGPARSER_H
+
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <vector>
+
+
+namespace argparser {
+    class Argument {
+        friend class ArgumentParser;
+
+    public:
+        std::string getDescription();
+        std::string getName();
+        Argument* setRequired(bool);
+        bool isFound();
+
+        // I have no fucking clue why this works.
+        template<typename T> T getValue() {
+            std::istringstream in(_get<std::string>());
+            T t = T();
+            in >> t >> std::ws;
+            return t;
+        }
+
+        Argument(std::string name, std::string description);
+
+    protected:
+        template<typename T> T _get() {
+            T t = T();
+            typename T::value_type vt;
+            for (auto &s : this->value) {
+                std::istringstream in(&s);
+                in >> vt;
+                t.push_back(vt);
+            }
+
+            return t;
+        }
+
+        std::string value;
+        bool _found{false};
```

Comment:
Which convention will we use for private variables? You sometimes use an underline as prefix.

Reply:
Good question

Reply:
I don't have a strong opinion about this, but I would choose no prefix. Then we do not need to refactor the code we have until now.